website/pages/component/[id]/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { GetStaticPaths, GetStaticProps } from 'next';
+import fs from 'fs';
+import path from 'path';
+import util from 'util';
+import { css } from 'emotion';
+import { GridContainer, GridItem } from '../../../components/grid/Grid';
+import Navigation from '../../../components/navigation/Navigation';
+import Header from '../../../components/layout/Header';
+import markdownToHtml from '../../../utils/markdownToHtml';
+import { BaseLayoutProps } from '../../../utils/types';
+
+const containerStyle = css`
+  margin-top: 12px;
+  display: flex;
+  flex-direction: row;
+  align-items: flex-start;
+`;
+
+const topMargin = css`
+  margin-top: 36px;
+`;
+
+export default function Component({
+  component,
+  changelog,
+  readme,
+}: BaseLayoutProps) {
+  console.log(component, changelog, readme);
+  return (
+    <div className={containerStyle}>
+      <Navigation />
+
+      <GridContainer justify="flex-start">
+        <GridItem md={6} lg={8}>
+          <div className={topMargin}>
+            <Header
+              component={component}
+              changelog={changelog}
+              readme={readme}
+            />
+          </div>
+        </GridItem>
+      </GridContainer>
+    </div>
+  );
+}
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  if (!params || typeof params.id !== 'string') {
+    return { props: {} };
+  }
+
+  const { id } = params;
+
+  const props: Partial<BaseLayoutProps> = { component: id };
+
+  const getFileContent = util.promisify(fs.readFile);
```

Comment:
We should promisify outside this method so we don't re-promisify every time the method is called.

Reply:
👍